### PR TITLE
Add MTParser and MT message models

### PR DESCRIPTION
**Pull request by sineththamuditha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/module/swiftiso20022/constants/ConnectorConstants.java [hunk 1]
```diff
@@ -74,13 +82,22 @@ private ConnectorConstants() {
     public static final String BLOCK03_VALIDATION_FLAG = "Validation Flag";
     public static final String BLOCK03_END_TO_END_REFERENCE = "End to End Reference";
     public static final String BLOCK03_SERVICE_TYPE_IDENTIFIER = "Service Type Identifier";
+    public static final String BLOCK03_MESSAGE_INPUT_REFERENCE = "Message Input Reference";
```
Comment (RivinduM) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/constants/ConnectorConstants.java:85]: The same constant is defined in L71. We can define these as common constants. No need of having the block no for the constant name.

Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/constants/ConnectorConstants.java]: Changed in commit 4daedbd4cd0fb94c36373fe83a8c13d9123a4efb

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/blocks/UserHeaderBlock.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field115;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field119;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field121;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field165;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field423;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field424;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field433;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field434;
+
+
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/blocks/UserHeaderBlock.java:35]: Lets remove extra line

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                    userHeaderBlock.setPaymentControlsInformation(Field434.parse(fields.get(key)));
+                    break;
+
+                // If tag didn't match any of the above tags, it is not allowed
+                default:
+
+                    String errorMessage = String.format(ConnectorConstants.ERROR_FIELD_NOT_ALLOWED_IN_BLOCK,
+                            key, ConnectorConstants.USER_HEADER_BLOCK);
+                    log.error(errorMessage);
+                    throw new MTMessageParsingException(errorMessage);
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java:169]: Shall we remove the extra lines inside the switch block

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                    ConnectorConstants.ERROR_BLOCK_INVALID_FORMAT, ConnectorConstants.TRAILER_BLOCK);
+            log.error(errorMessage);
+            throw new MTMessageParsingException(errorMessage);
+        }
+
+        TrailerBlock trailerBlock = new TrailerBlock();
+
+        for (String key : fields.keySet()) {
+
+            switch (key) {
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java:205]: Same comment https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/11/files#r1574133144

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MTParserUtils.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        // Get matcher to the regex matching -> {(Tag):(Field Value)}
+        Matcher matcher = MTParserConstants.CURLY_BRACKETS_FIELDS_MATCHING_PATTERN.matcher(fieldsString);
+
+        while (matcher.find()) {
+
+            // group 1 -> Tag
+            // group 2 -> Field Value
+            fields.put(matcher.group(1), matcher.group(2));
+
+        }
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MTParserUtils.java:56]: Remove extra lines

File: src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    public void testParseUserHeaderBlock(String userHeaderBlockString) throws MTMessageParsingException {
+
+        MTParser.parse(Map.of(ConnectorConstants.USER_HEADER_BLOCK_KEY, userHeaderBlockString), new MTMessage());
+    }
+
+    @Test(expectedExceptions = MTMessageParsingException.class, dataProvider = "invalidTrailerBlockDataProvider",
+            dataProviderClass = MTParserTestConstants.class)
+    public void testParseTrailerBlock(String trailerBlockString) throws MTMessageParsingException {
+
+        MTParser.parse(Map.of(ConnectorConstants.TRAILER_BLOCK_KEY, trailerBlockString), new MTMessage());
```
Comment (Ashi1993) [inline src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java:57]: Shall we add tests for negative scenarios as well

File: src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    }
```
Comment (Ashi1993) [inline src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java:58]: How do you confirm whether the expected error is thrown for each scenario?

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/blocks/UserHeaderBlock.java (hunk 2 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/blocks/UserHeaderBlock.java]: Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java (hunk 3 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java:268]: Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java (hunk 4 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java:303]: Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MTParserUtils.java (hunk 5 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MTParserUtils.java:151]: Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0

File: src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java (hunk 7 above)
Comment (sineththamuditha) [reply src/test/java/org/wso2/carbon/module/swiftiso20022/MTParserTests.java:257]: Units tests are added to check each error message for each field in a82da3f30afc3995c983ec8d68a04acfc363b6c0

**Review by Ashi1993 — APPROVED** [review]

**Review by RivinduM — APPROVED** [review]
